ide/lsp.client/src/org/netbeans/modules/lsp/client/LSPBindings.java

```diff
@@ -269,6 +280,12 @@ private static LSPBindings buildBindings(ServerDescription inDescription, Projec
 
         for (LanguageServerProvider provider : MimeLookup.getLookup(mt).lookupAll(LanguageServerProvider.class)) {
             final Lookup lkp = prj != null ? Lookups.fixed(prj, mimeTypeInfo, restarter) : Lookups.fixed(mimeTypeInfo, restarter);
+            inDescription.mimeTypes = Collections.singleton(mt);
+            // If this is a MultiMimeLanguageServerProvider, then retrieve all 
+            // mime types handled by this server.
+            if (provider instanceof MultiMimeLanguageServerProvider) {
+                inDescription.mimeTypes = ((MultiMimeLanguageServerProvider)provider).getMimeTypes();
+            }
```

Comment:
I would make a defensive copy in line 283, creating a new HashSet at that point from the core mimetype and adding the other mimetypes to that. Else the provider could change the backing set after the description was used for starting, causing an imbalance (yes not really realistic, but ...).

Reply:
Thanks for the tip!